ogusa/utils.py

```diff
@@ -701,3 +706,128 @@ def read_cbo_forecast():
+        total (int>=1): how many tasks are in the job
+        source_name (string): name of source data
+        prefix (string): what to print before the progress bar
+        suffix (string): what to print after the progress bar
+        decimals (int>=0): how many decimals in the percentage
+        bar_length (int>=3): how many boxes in the progress bar
+
+    Functions called: None
+
+    Objects created within function:
+        str_format (string): string containing percentage completed
+        percents (string): percentage completed
+        filled_length (int): number of boxes in the progress bar to fill
+        bar (string): progress bar
+
+    Returns: None
+    '''
+    str_format = "{0:." + str(decimals) + "f}"
+    percents = str_format.format(100 * (iteration / float(total)))
+    filled_length = int(round(bar_length * iteration / float(total)))
+    bar = '█' * filled_length + '-' * (bar_length - filled_length)
+
+    if iteration == 0:
+        if source_name == '':
+            sys.stdout.write('Accessing data files...\n')
+        else:
+            sys.stdout.write('Accessing ' + source_name +
+                             ' data files...\n')
+
+    sys.stdout.write('\r%s |%s| %s%s %s' %
+                     (prefix, bar, percents, '%', suffix)),
+
+    if iteration == total:
+        sys.stdout.write('\n')
+        sys.stdout.write('Computing...\n')
+    sys.stdout.flush()
+
+
+def fetch_files_from_web(file_urls):
+    '''
+    Fetches zip files from respective web addresses and saves them as
+    temporary files. Prints progress bar as it downloads the files.
+
+    Args:
+        file_urls (list of strings): list of URLs of respective data zip
+            files
+
+    Functions called:
+        print_progress()
+
+    Objects created within function:
+        local_paths = list, local paths for teporary files
+        iteration   = int, the number of files that have been downloaded
+        total       = total, the total number of files to download
+        f           = temporary file of monthly CPS survey
+        path        = string, local path for temporary file
+        zipped_file = ZipFile object, opened zipfile
+
+    Files created by this function:
+        .dta file for each year of SCF data
+
+    Returns:
+        local_paths (list of strings): local paths of temporary data
+            files
+    '''
+    local_paths = []
+
+    iteration = 0
+    total = len(file_urls)
+    print_progress(iteration, total, source_name='SCF')
+
+    for file_url in file_urls:
+        # url = requests.get(file_url) (if using reuests package)
+        url = urllib.request.urlopen(file_url)
+
+        f = NamedTemporaryFile(delete=False)
+        path = f.name
+
+        # url.content (if using requests package)
+        with ZipFile(BytesIO(url.read())) as zipped_file:
+            for contained_file in zipped_file.namelist():
+                for line in zipped_file.open(contained_file).readlines():
```

Comment:
Can you just do a `.read()` instead of `.readlines()` to avoid this loop?

Reply:
@jdebacker . Not sure. I will test `.read()` here. I am not sure what the `.readlines()` loop line was doing in my [`hrs_by_age.py`](https://github.com/OpenSourceEcon/CPS_hrs_age/blob/master/hrs_by_age.py) code.

Reply:
@jdebacker . I took this loop out and just used the `.read()` method and it seems to pass all the tests locally on my machine, including the correct moments. If this passes the CI and codecov tests on this repo, then I think it is ready for your review and merge.